contracts/src/components/labor_auction.cairo

```diff
@@ -0,0 +1,204 @@
+/// decay constant, and per-time-unit rate.
+#[derive(Copy, Drop, Serde, starknet::Storage)]
+struct LinearVRGDA {
+    target_price: Fixed,
+    decay_constant: Fixed,
+    per_time_unit: Fixed,
+}
+
+#[generate_trait]
+impl LinearVRGDAImpl of LinearVRGDATrait {
+    /// Calculates the target sale time based on the quantity sold.
+    ///
+    /// # Arguments
+    ///
+    /// * `sold`: Quantity sold.
+    ///
+    /// # Returns
+    ///
+    /// * A `Fixed` representing the target sale time.
+    fn get_target_sale_time(self: @LinearVRGDA, sold: Fixed) -> Fixed {
+        sold / *self.per_time_unit
+    }
+
+    /// Calculates the VRGDA price at a specific time since the auction started.
+    ///
+    /// # Arguments
+    ///
+    /// * `time_since_start`: Time since the auction started.
+    /// * `sold`: Quantity sold.
+    ///
+    /// # Returns
+    ///
+    /// * A `Fixed` representing the price.
+    fn get_vrgda_price(self: @LinearVRGDA, time_since_start: Fixed, sold: Fixed) -> Fixed {
+        *self.target_price
+            * exp(
+                *self.decay_constant
+                    * (time_since_start
+                        - self.get_target_sale_time(sold + FixedTrait::new(1, false)))
+            )
+    }
+
+    fn get_reverse_vrgda_price(self: @LinearVRGDA, time_since_start: Fixed, sold: Fixed) -> Fixed {
+        *self.target_price
+            * exp(
+                (*self.decay_constant * FixedTrait::new(1, true))
+                    * (time_since_start
+                        - self.get_target_sale_time(sold + FixedTrait::new(1, false)))
+            )
+    }
+}
+
+
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct LaborAuction {
+    #[key]
+    zone: u8,
+    decay_constant_mag: u128,
+    decay_constant_sign: bool,
+    per_time_unit: u128,
+    start_time: u64,
+    sold: u128,
+}
+
+#[generate_trait]
+impl LaborAuctionImpl of LaborAuctionTrait {
+    fn to_LinearVRGDA(self: LaborAuction) -> LinearVRGDA {
+        LinearVRGDA {
+            target_price: FixedTrait::new_unscaled(1, false),
+            decay_constant: ln(
+                FixedTrait::ONE()
+                    - FixedTrait::new(self.decay_constant_mag, self.decay_constant_sign)
+            ),
+            per_time_unit: FixedTrait::new_unscaled(self.per_time_unit, false)
+        }
+    }
+
+    fn get_price(self: LaborAuction) -> Fixed {
+        // time since auction start
+        let time_since_start: u128 = get_block_timestamp().into() - self.start_time.into();
+        // get current price
+        self
+            .to_LinearVRGDA()
+            .get_vrgda_price(
+                // 1 period = 1 day
+                FixedTrait::new_unscaled(time_since_start / 86400, false), // time since start
+                FixedTrait::new_unscaled(self.sold, false), // amount sold
+            )
+    }
+
+    fn sell(ref self: LaborAuction, world: IWorldDispatcher) {
```

Comment:
The world dispatcher isn't doing anything

Reply:
oups, totally right